src/manager/manager.py

```diff
@@ -36,16 +36,22 @@ def manager_event_loop(manager: Manager, vehicles: list[Vehicle], time: float):
     
 def _update_manager_vehicle_list(manager: Manager, vehicles: list[Vehicle]):
     for vehicle in vehicles:
-        # vehicle already in list?
-        vehicle_in_list = any(manager_vehicle.id == vehicle.id for manager_vehicle in manager.vehicles)
-        if vehicle_in_list: continue
 
-        # vehicle within manager radius?
+        # vehicle within manager radius? 
         distance_to_vehicle = np.linalg.norm(route_position_to_world_position(vehicle.route, vehicle.route_position)-manager.position)
-        if distance_to_vehicle > manager.radius: continue
 
-        # vehicle is not in list and within radius, add to list
-        manager.vehicles.append(vehicle)
+        # vehicle already in list and within manager radius?
+        vehicle_in_list = any(manager_vehicle.id == vehicle.id for manager_vehicle in manager.vehicles)
+
+        # append if not in list and inside radius
+        if not vehicle_in_list and distance_to_vehicle <= manager.radius: 
+            manager.vehicles.append(vehicle)
+            continue
+
+        # remove if in list and outside radius
+        if vehicle_in_list and distance_to_vehicle > manager.radius: 
+            manager.vehicles.remove(vehicle)
+            continue
```

Comment:
amazing, good work
think we can drop the 'continue' and make it a normal if else case.